File: 6-1.py
#!/usr/bin/env python

fishes = [int(fish) for fish in open('6.input').readlines()[0].split(",")]

max_days = 80

for day in range(1, max_days + 1):
    next_fishes = []
    for fish in fishes:
        if fish > 0:
            next_fishes.append(fish - 1)
        else:
            next_fishes.append(6)
            next_fishes.append(8)
    fishes = next_fishes

print(len(fishes))

File: 6-1.rb
#!/usr/bin/env ruby

fishes = File.read('6.input').strip.split(",").map(&:to_i)

max_days = 80

1.upto(max_days) do |day|
  next_fishes = []
  fishes.each do |fish|
    if fish > 0
      next_fishes.append fish - 1
    else
      next_fishes.append 6
      next_fishes.append 8
    end
  end
  fishes = next_fishes
end

print fishes.size, "\n"
